ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java

```diff
@@ -266,6 +266,13 @@ default void notifySnapshotInstalled(InstallSnapshotResult result, long snapshot
      * Notify the {@link StateMachine} that a raft server has step down.
      */
     default void notifyServerShutdown(RoleInfoProto roleInfo) {}
```

Comment:
We should deprecate the old method and change its default to
```java
    /**
     * The same as notifyServerShutdown(roleInfo, false).
     *
     * @deprecated Implementations should override {@link #notifyServerShutdown(RoleInfoProto, boolean)} instead.
     */
    @Deprecated
    default void notifyServerShutdown(RoleInfoProto roleInfo) {
      notifyServerShutdown(roleInfo, false);
    }
```
Could you also update the caller?
```java
+++ b/ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java
@@ -334,7 +334,7 @@ class LeaderElection implements Runnable {
         case NOT_IN_CONF:
         case SHUTDOWN:
           server.close();
-          server.getStateMachine().event().notifyServerShutdown(server.getRoleInfoProto());
+          server.getStateMachine().event().notifyServerShutdown(server.getRoleInfoProto(), false);
           return false;
         case TIMEOUT:
```

Reply:
Sure. Addressed in new commit.